Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3259,6 +3285,37 @@ void MediaPlayerPrivateGStreamer::elementSetupCallback(MediaPlayerPrivateGStream
 #endif // USE(WESTEROS_SINK)
 }
 
+OdhDrm MediaPlayerPrivateGStreamer::AvContextGetterImpl::getDrm()
+{
+    GST_DEBUG("[ODH_REPORT_AVPIPELINE] getDrm");
+    return OdhDrm::NONE;
+}
+
+OdhOwner MediaPlayerPrivateGStreamer::AvContextGetterImpl::getOwner()
+{
+    GST_DEBUG("[ODH_REPORT_AVPIPELINE] getOwner");
+    return OdhOwner::WPE;
+}
+
+GstElement* MediaPlayerPrivateGStreamer::AvContextGetterImpl::getDecoder(const OdhMediaType /*media_type*/)
+{
+    GST_DEBUG("[ODH_REPORT_AVPIPELINE] getDecoder");
+    return nullptr;
+}
+
+GstElement* MediaPlayerPrivateGStreamer::AvContextGetterImpl::getPipeline()
+{
+    GST_DEBUG("[ODH_REPORT_AVPIPELINE] getPipeline");
+    return m_pipeline.get();
+}
+
+std::string MediaPlayerPrivateGStreamer::AvContextGetterImpl::getCodec(const OdhMediaType /*media_type*/)
+{
+    GST_DEBUG("[ODH_REPORT_AVPIPELINE] getCodec");
+    return "";
```

Comment:
why it's not returng anything?

Reply:
In case of WPE the codec is taken from the pipeline by the libodhott. We have to leave the old API for back compatibility(netflix still uses it)

Reply:
You re right, we can. I didn' do that to minimize changes in libodhott API but it seems to be a better solution for future changes.